jax/experimental/jax2tf/tests/primitive_harness.py

```diff
@@ -405,6 +407,120 @@ def parameterized(harness_group: Iterable[Harness],
+                                (FftType.IRFFT, ((shape[-1] - 1) * 2,)),
+                                (FftType.IRFFT, [])]
+) + tuple( # 2D tests, no complex
+  Harness(f"2d_shape={jtu.format_shape_dtype_string(shape, dtype)}_ffttype={fft_type}_fftlengths={fft_lengths}",
+          lax.lax_fft.fft,
+          [RandArg(shape, dtype), StaticArg(fft_type), StaticArg(fft_lengths)],
+          shape=shape,
+          dtype=dtype,
+          fft_type=fft_type,
+          fft_lengths=fft_lengths)
+  for dtype in [t for t in jtu.dtypes.all if t not in jtu.dtypes.complex]
+  for shape in [(12, 13), (14, 15, 16)]
+  for fft_type, fft_lengths in [(FftType.FFT, shape[-2:]),
+                                (FftType.IFFT, shape[-2:]),
+                                (FftType.RFFT, shape[-2:]),
+                                (FftType.IRFFT, shape[-2:-1] + ((shape[-1] - 1) * 2,))]
+) + tuple( # 2D tests, complex
+  Harness(f"2d_shape={jtu.format_shape_dtype_string(shape, dtype)}_ffttype={fft_type}_fftlengths={fft_lengths}",
+          lax.lax_fft.fft,
+          [RandArg(shape, dtype), StaticArg(fft_type), StaticArg(fft_lengths)],
+          shape=shape,
+          dtype=dtype,
+          fft_type=fft_type,
+          fft_lengths=fft_lengths)
+  for dtype in jtu.dtypes.complex
+  for shape in [(12, 13), (14, 15, 16)]
+  for fft_type, fft_lengths in [(FftType.FFT, shape[-2:]),
+                                (FftType.IFFT, shape[-2:]),
+                                (FftType.IRFFT, shape[-2:-1] + ((shape[-1] - 1) * 2,))]
+) + tuple( # 3D tests, no complex
+  Harness(f"3d_shape={jtu.format_shape_dtype_string(shape, dtype)}_ffttype={fft_type}_fftlengths={fft_lengths}",
+          lax.lax_fft.fft,
+          [RandArg(shape, dtype), StaticArg(fft_type), StaticArg(fft_lengths)],
+          shape=shape,
+          dtype=dtype,
+          fft_type=fft_type,
+          fft_lengths=fft_lengths)
+  for dtype in [t for t in jtu.dtypes.all if t not in jtu.dtypes.complex]
+  for shape in [(14, 15, 16)]
+  for fft_type, fft_lengths in [(FftType.FFT, shape[-3:]),
+                                (FftType.IFFT, shape[-3:]),
+                                (FftType.RFFT, shape[-3:]),
+                                (FftType.IRFFT, shape[-3:-1] + ((shape[-1] - 1) * 2,))]
+) + tuple( # 3D tests, complex
+  Harness(f"3d_shape={jtu.format_shape_dtype_string(shape, dtype)}_ffttype={fft_type}_fftlengths={fft_lengths}",
+          lax.lax_fft.fft,
+          [RandArg(shape, dtype), StaticArg(fft_type), StaticArg(fft_lengths)],
+          shape=shape,
+          dtype=dtype,
+          fft_type=fft_type,
+          fft_lengths=fft_lengths)
+  for dtype in jtu.dtypes.complex
+  for shape in [(14, 15, 16)]
+  for fft_type, fft_lengths in [(FftType.FFT, shape[-3:]),
+                                (FftType.IFFT, shape[-3:]),
+                                (FftType.IRFFT, shape[-3:-1] + ((shape[-1] - 1) * 2,))]
+) + tuple( # 4D test, no complex, should error
```

Comment:
Perhaps we can merge the non-complex and complex 4D tests?

Reply:
Merged all the complex/non-complex tests together.